src/arkreactor/Utils.cpp

```diff
@@ -29,4 +29,32 @@ namespace Ark::Utils
         }
         return digit_places;
     }
+
+    int levenshteinDistance(const std::string& str1, const std::string& str2)
+    {
+        int str1_len = str1.size();
+        int str2_len = str2.size();
+        std::vector<std::vector<int>> edit_distances(str1_len + 1, std::vector<int>(str2_len + 1, 0));
+
+        for (int i = 0; i < str1_len + 1; i++)
+            edit_distances[i][0] = i;
+
+        for (int j = 0; j < str2_len + 1; j++)
+            edit_distances[0][j] = j;
+
+        for (int i = 1; i < str1_len + 1; i++)
+        {
+            for (int j = 1; j < str2_len + 1; j++)
+            {
+                int indicator = str1[i - 1] == str2[j - 1] ? 0 : 1;
+                edit_distances[i][j] = std::min({
+                    edit_distances[i - 1][j] + 1,             // deletion
+                    edit_distances[i][j - 1] + 1,             // insertion
+                    edit_distances[i - 1][j - 1] + indicator  // substitution
+                });
+            }
+        }
+
+        return edit_distances[str1_len][str2_len];
+    }
```

Comment:
AFAICT, you could use `std::size_t` instead of `int`, as this will prevent downcasting.